src/Collector.cs

```diff
@@ -79,25 +83,34 @@ private void StartBackgroundWorker()
 
     private void WorkerOnDoWork(object sender, DoWorkEventArgs doWorkEventArgs)
     {
+      var failures = 0;
+      Socket socket = null;
+
+      // This is not really the worker loop, it is just very careful connect/authenticate
+      // If you want the worker loop, look at SendQueuedMessages
       while (!_worker.CancellationPending)
         {
-          Socket socket = null;
-          var failures = 0;
-
           try
             {
               socket = Connect();
-              Authenticate(socket);
-              failures = 0;
+              if(!Authenticate(socket))
+                {
+                  socket.Close();
+                  Thread.Sleep(5000);
```

Comment:
perhaps some explanation of why there is a sleep here is in order?

stampede prevention?